### ISSUE TITLE
Give higher importance to config than env variables

### ISSUE THREAD

**Issue by meain** [Peltoche/lsd#546]
Flip the order here and make config before env https://github.com/meain/lsd/blob/70ec1d6c51d209ea3e29ba2ae8208bbbc45690b2/src/flags.rs#L118-L124 .

This issue came up from https://github.com/Peltoche/lsd/pull/544 . We primary wanted to do this switch becase this will help us comply with how https://no-color.org expect the flag to work. Also, this actually makes more sense as if someone would like to configure lsd separately instead of using a global env value they should be able to do so in the config file and if they wanna configure it using env value, they can always just skip the config.

This currently has one small blocker in that we use a default config if the user does not have a config and so technically we "always" have a config if the user does not have a config.

**Comment by heyxsh** [comment]
I agree. The config file should be prioritize first. It took me a while to figure out why my config wasn't read.

**Comment by meain** [comment]
I am not sure if this is the right choice anymore. While it makes sense for `NO_COLOR`, I don't think it makes sense for most of the other env variables.

**Comment by zwpaper** [comment]
I also think env variables should have a higher priority than config files, it is that in most others cases I knew.

Env variables are easier to modify than config files.
